server/api/login.js

```diff
@@ -95,11 +106,42 @@ internals.applyRoutes = function (server, next) {
                         return reply(session);
                     });
                 }
+            }, {
+                assign: 'validSession',
+                method: function (request, reply) {
+
+                    const existingSession = request.pre.existingSession;
+
+                    if (!existingSession) {
+
+                        const userAgent = Useragent.lookup(request.headers['user-agent']);
+                        const ip = request.info.remoteAddress;
+                        const browser = userAgent.family;
+                        const os = userAgent.os.toString();
+
+                        Session.create(request.pre.user._id.toString(), ip, browser, os, (err, session) => {
+
+                            if (err) {
+                                return reply(err);
+                            }
+
+                            return reply(session);
+                        });
+                    }
+                    else if (existingSession.revoked) {
+
+                        return reply(Boom.forbidden('Session is revoked'));
```

Comment:
End sentence with a period.

Ex: `return done(Boom.forbidden('Session is revoked.'));`

Reply:
I'm not sure what's better; keeping the revoked session or just deleting them when they're revoked.

I think if we just delete the session, it would turn into a _session not found_, and we shouldn't to handle these cases.

Reply:
If the session is deleted then, they would be able to log in again. Should Revoking mean logging out or block? With the way it is implemented now, we can change the boolean in session from revoked to block.